apps/mark-scan/backend/src/custom-paper-handler/state_machine.ts

```diff
@@ -324,15 +369,87 @@ export function buildMachine(
             actions: assign({
-              interpretation: (_, event) => {
-                return event.data;
-              },
+              interpretation: (_, event) => event.data,
             }),
           },
         },
       },
+      // Intermediate state to conditionally transition based on ballot interpretation
+      transition_interpretation: {
+        always: [
+          {
+            target: 'presenting_ballot',
+            cond: (context) =>
+              context.interpretation
+                ? context.interpretation[0].interpretation.type ===
+                  'InterpretedBmdPage'
+                : false,
+          },
+          {
+            target: 'blank_page_interpretation',
+            cond: (context) =>
+              context.interpretation
+                ? context.interpretation[0].interpretation.type === 'BlankPage'
+                : false,
+          },
+        ],
+      },
+      blank_page_interpretation: {
+        invoke: pollPaperStatus(),
+        entry: async (context) => {
+          await context.driver.presentPaper();
+        },
+        initial: 'presenting_paper',
+        // These nested states differ slightly from the top-level paper load states so we can't reuse the latter.
+        states: {
+          presenting_paper: {
+            // Heavier paper can fail to eject completely and trigger a jam state even though the paper isn't physically jammed.
+            // To work around this, we avoid ejecting to front. Instead, we present the paper so it's held by the device in a
+            // stable non-jam state. We instruct the poll worker to remove the paper directly from the 'presenting' state.
+            entry: async (context) => await context.driver.presentPaper(),
+            on: { NO_PAPER_ANYWHERE: 'accepting_paper' },
+          },
+          accepting_paper: {
+            on: {
+              PAPER_READY_TO_LOAD: 'load_paper',
+            },
+          },
+          load_paper: {
+            entry: async (context) => {
+              await context.driver.loadPaper();
+              await context.driver.parkPaper();
+            },
+            on: {
+              PAPER_PARKED: 'done',
+              NO_PAPER_ANYWHERE: 'accepting_paper',
+            },
+          },
+          done: {
+            type: 'final',
+            entry: () => {
+              assign({
+                interpretation: undefined,
+                scannedImagePaths: undefined,
+              });
+            },
+          },
+        },
+        onDone: { target: 'paper_reloaded' },
+      },
+      // `paper_reloaded` could be a substate of `blank_page_interpretation` but by keeping
+      // them separate we can avoid exposing all the substates of `blank_page_interpretation`
+      // to the frontend.
+      paper_reloaded: {
+        invoke: pollAuthStatus(),
+        entry: () => {
+          debug('Placeholder paper_reloaded');
+        },
```

Comment:
Remove debug?

Reply:
Sorry, I think I cleaned this up but accidentally overwrote with a `git reset` before committing.